### ISSUE TITLE
Using AMD RYZEN

### ISSUE THREAD

**Issue by robertofuentesr** [facebookresearch/llama#745]
Hi everyone!

I am trying to run the 7 B chat model faster, I am currently using only my nvidia card, but I also have an AMD RYZEN card, is there a way so I can use both memory cards?
The model runs too slow right now, Any help I appreciate beforehand, this is my card characteristic:

Card name: AMD Radeon(TM) Graphics
Manufacturer: Advanced Micro Devices, Inc.
Chip type: AMD Radeon Graphics Processor (0x1681)
DAC type: Internal DAC(400MHz)
Device Type: Full Device (POST)

Device Problem Code: No Problem
Driver Problem Code: Unknown
Display Memory: 16456 MB
Dedicated Memory: 485 MB
Shared Memory: 15970 MB

**Comment by jspisak** [comment]
Hi Robert,
Thanks for entering this issue. Our intent is not to support beyond Nvidia GPUs but I know the AMD team and others have support for their GPUs. I am looping in @sunway513 and @jithunnair-amd who might know which code base is best to work from. 

Once one of them responds, I'll close this issue and also update the readme so we are more clear about what is supported and what isn't. Thanks!

**Comment by hongxiayang** [comment]
@robertofuentesr  Before we can see whether you can do NVIDIA and AMD in one training/inference workflow, we need to check whether you can just use AMD exclusively first to train on your AMD card. Please let me know whether you need help to do this step.

**Comment by hongxiayang** [comment]
@robertofuentesr  To install PyTorch on ROCM, follow these steps:
(1) Install ROCm, check https://rocm.docs.amd.com/en/latest/deploy/linux/quick_start.html
(2) Recommend you try to build based on ROCm's docker image (Option 1). 
Instruction is as below:
https://github.com/ROCmSoftwarePlatform/pytorch/wiki/Building-PyTorch-for-ROCm#option-1-recommended--use-docker-image-with-pytorch-pre-installed

Thanks.

**Comment by robertofuentesr** [comment]
@hongxiayang  I am trying to use ROCm, but I haven't had any luck yet :/ 
I did install it though, but I can't find my AMD card yet. 
I will continue to try to see if I get any luck.

**Comment by hongxiayang** [comment]
> @hongxiayang I am trying to use ROCm, but I haven't had any luck yet :/ I did install it though, but I can't find my AMD card yet. I will continue to try to see if I get any luck.

Ok, @robertofuentesr  you can also post with details with your PyTorch on ROCm installation/setup/verification in pytorch/pytorch repo's issues (https://github.com/pytorch/pytorch/issues). Please run "rocminfo|grep gfx", so that we know whether your card is aligned with the requirement of the latest pytorch version.

**Comment by robertofuentesr** [comment]
@hongxiayang 
I try this:
rocminfo|grep gfx
But it didn't recognize the first command, I guess the installation was unsuccessful.

I also noticed this on the official website of Pytorch:  It looks like window don't have ROCm
![image](https://github.com/facebookresearch/llama/assets/53060733/6a242be7-cbc9-4963-8450-dcb9d89dff51).

I am gonna learn a little about dockers and try the second approach you gave me above.
Just in case, do you know if there is a conda command to get ROCm installed with pytorch  2.0.1? 


 thanks for all your help!

**Comment by hongxiayang** [comment]
@robertofuentesr  Correct, Windows Support for PyTorch on ROCm is currently not available. This includes WSL2 based workflows.
